contracts/SynthetixRouterStrategy.sol

```diff
@@ -0,0 +1,311 @@
+import {
+    SafeERC20,
+    SafeMath,
+    IERC20,
+    Address
+} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+import "./Interfaces/erc20/IERC20Extended.sol";
+
+import "./RouterStrategy.sol";
+import "./Synthetix.sol";
+
+interface IUni {
+    function getAmountsOut(uint256 amountIn, address[] calldata path)
+        external
+        view
+        returns (uint256[] memory amounts);
+}
+
+contract SynthetixRouterStrategy is RouterStrategy, Synthetix {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    uint256 internal constant DUST_THRESHOLD = 10_000;
+    address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant uniswapRouter =
+        0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+
+    constructor(
+        address _vault,
+        address _yVault,
+        bytes32 _synth,
+        string memory _strategyName
+    ) public RouterStrategy(_vault, _yVault, _strategyName) {
+        _initializeSynthetix(_synth);
+    }
+
+    event Cloned(address indexed clone);
+
+    function cloneRouter(
+        address _vault,
+        address _strategist,
+        address _rewards,
+        address _keeper,
+        address _yVault,
+        bytes32 _synth,
+        string memory _strategyName
+    ) external returns (address newStrategy) {
+        require(isOriginal);
+        // Copied from https://github.com/optionality/clone-factory/blob/master/contracts/CloneFactory.sol
+        bytes20 addressBytes = bytes20(address(this));
+        assembly {
+            // EIP-1167 bytecode
+            let clone_code := mload(0x40)
+            mstore(
+                clone_code,
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newStrategy := create(0, clone_code, 0x37)
+        }
+
+        SynthetixRouterStrategy(newStrategy).initialize(
+            _vault,
+            _strategist,
+            _rewards,
+            _keeper,
+            _yVault,
+            _synth,
+            _strategyName
+        );
+
+        emit Cloned(newStrategy);
+    }
+
+    function initialize(
+        address _vault,
+        address _strategist,
+        address _rewards,
+        address _keeper,
+        address _yVault,
+        bytes32 _synth,
+        string memory _strategyName
+    ) public {
+        _initialize(_vault, _strategist, _rewards, _keeper);
+        require(address(yVault) == address(0));
+        _initializeSynthetix(_synth);
+        _initializeThis(_yVault, _strategyName);
+    }
+
+    function name() external view override returns (string memory) {
```

Comment:
this doesn't make sense, your clone method receives `_strategyName` which the parent class uses correctly in the `name()`. You might be able to avoid overriding this method.